####-sparse-observable.md

```diff
@@ -0,0 +1,517 @@
+If it shares that role with `SparsePauliOp`, nothing substantial changes about this RFC.
+That is the hypothetical Estimator pub would be a three-item record type:
+
+```python
+class EstimatorPub:
+    circuit: QuantumCircuit
+    observables: ND[SparseObservable]
+    bindings: ND[Mapping[Parameter, Value]]
+```
+
+where `ND` is a stand-in for a type supplying arbitrary-dimensioned n-dimensional broadcasting and slicing, such as a Numpy array of `object`, or a custom container defined in the primitives base classes.
+The particulars of `Parameter` and `Value` and the containing mapping are irrelevant to this RFC.
+
+`SparseObservable` will support some set of mathematical operations.
+At a minimum, the following will be supported:
+
+* addition of two `SparseObservable`s
+* tensor product of two `SparseObservable`s
+* evolution of one `SparseObservable` by a Pauli string ($A' = P A P^\dagger$ for `SparseObservable` $A$ and Pauli $P$)
+* multiplication by complex scalars
+* structural equality of two `SparseObservable`s (structural not mathematical; it's highly inefficient to detect equality if the abstract terms form an over-complete spanning set).
+
+See "Questions" for more on the mathematical-operator support.
+
+Iterator methods for accessing the terms and their coefficients will be supported.
+The exact format of these is to be informed by the intended use cases, and there can be more than one iterator method available.
+
+
+### Representation of measurement bases
+
+> [!NOTE]
+> **Summary**: Qiskit will supply some helper functions on `SparseObservable`, but for MVP 1 of the primitive internals, we can keep using `PauliList` for this bit, and we can optimise it in the future if needed.
+
+While the user input is a n-dimensional group of observable sums, the actual execution is serialised and only cares about measurement-basis transformation needed for any given multi-qubit term.
+See the "Detailed Design" section for more discussion on exactly how this grouping works.
+
+From Python-space, then, we produce an iterable of "preparation programs" (pairs of `(circuit, bindings)`), and for element of that we have an iterable of `SparseObservable` with we want to map to a group of groups of measurement bases.
+The iterable of preparation programs is produced in Python-space by n-dimensional slicing/broadcasting tricks.
+
+A core function of `Estimator` is to group terms that can be measured within the same execution.
+There are many ways to do this, and we do not want to tie the observable to one particular implementation.
+Qiskit will provide a function `SparseObservable.measurement_bases(*observables)` that takes an arbitrary number of `SparseObservable` instances and returns a set of the measurement bases needed to measure all terms.
```

Comment:
This method would then just give all required measurement basis w/o any kind of optimization, which is left to other algorithms, right?

Reply:
Yeah, the purpose is to separate "grouping of measurement bases" (which primitives / users will likely want to configure) from "what are the measurement bases?".